firedrake/slate/static_condensation/hybridization.py

```diff
@@ -250,6 +250,13 @@ def initialize(self, pc):
         trace_ksp.setDMActive(False)
         trace_ksp.setOptionsPrefix(prefix)
         trace_ksp.setOperators(Smat, Smat)
+
+        # Option to add custom monitor
+        monitor = self.ctx.appctx.get('custom_monitor', None)
+        if monitor:
+            monitor.add_reconstructor(self.backward_substitution)
```

Comment:
What is this magic `add_reconstructor` function? This looks tremendously specific.

Reply:
It's a way to get hooks to the backward substitution so that we can monitor the error in the eliminated variables during the trace solve. At this point we could:
a) remove it, but we are using it, so we'd need to just do this on a branch, I suppose
b) find some way to document this add_reconstructor function. I suppose that we just need to say in the documentation what things should be added to the appctx?

Reply:
OK, fine. I would prefer that any putative redesign thinks properly about what that interface should be.